qiskit/transpiler/passes/utils/gates_basis.py

```diff
@@ -13,27 +13,51 @@
 """Check if all gates in the DAGCircuit are in the specified basis gates."""
 
 from qiskit.transpiler.basepasses import AnalysisPass
+from qiskit.transpiler.exceptions import TranspilerError
 
 
 class GatesInBasis(AnalysisPass):
     """Check if all gates in a DAG are in a given set of gates"""
 
-    def __init__(self, basis_gates):
+    def __init__(self, basis_gates=None, target=None):
         """Initialize the GatesInBasis pass.
 
         Args:
             basis_gates (list): The list of strings representing the set of basis gates.
+            target (Target): The target representing the backend. If specified
+                this will be used instead of the ``basis_gates`` parameter
+
+        Raises:
+            TranspilerError: If neither basis_gates or target is set.
         """
         super().__init__()
-        self._basis_gates = set(basis_gates)
+        if basis_gates is None and target is None:
+            raise TranspilerError(
+                "A value for 'basis_gates' or 'target' must be set to use this pass"
+            )
+        if basis_gates is not None:
```

Comment:
Is it worth raising an error if both target and basis gates are specified, or perhaps just checking both? Not sure if someone might expect the union of those things to be used to answer the question if they don't read the description.

Reply:
We could do this I guess, I normally have been avoiding a hard failure in this case mostly because how passes are used in the preset pass managers we need to pass both basis_gates and target to each passes' constructor as we don't know if we're going to have a backendv1 or backendv2. So a typical application for the preset pass managers is to have something like:

```
def build_passmanager(pass_manager_config):
target = pass_manager_config.target
basis_gates = pass_manager_config.basis_gates

GatesInBasis(basis_gates=basis_gates, target=target)
```

in the BackendV1 case target is None and in v2 both basis gates and target are defined. It's simpler to just make target higher priority in the pass then adding a conditional around each pass in the pass manager build. That being said this pass isn't actually being used in the preset pass managers yet (there is a PR looking at doing it #6940 though)